x-pack/legacy/plugins/spaces/server/lib/request_interceptors/on_request_interceptor.ts

```diff
@@ -7,26 +7,25 @@ import {
   KibanaRequest,
   OnPreAuthToolkit,
   LifecycleResponseFactory,
-  HttpServiceSetup,
+  CoreSetup,
 } from 'src/core/server';
-import { KibanaConfig } from 'src/legacy/server/kbn_server';
 import { format } from 'url';
 import { DEFAULT_SPACE_ID } from '../../../common/constants';
 import { getSpaceIdFromPath } from '../spaces_url_parser';
 import { modifyUrl } from '../utils/url';
+import { LegacyAPI } from '../../new_platform/plugin';
 
 export interface OnRequestInterceptorDeps {
-  config: KibanaConfig;
-  http: HttpServiceSetup;
+  getLegacyAPI(): LegacyAPI;
+  http: CoreSetup['http'];
 }
-export function initSpacesOnRequestInterceptor({ config, http }: OnRequestInterceptorDeps) {
-  const serverBasePath: string = config.get('server.basePath');
-
+export function initSpacesOnRequestInterceptor({ getLegacyAPI, http }: OnRequestInterceptorDeps) {
   http.registerOnPreAuth(async function spacesOnPreAuthHandler(
     request: KibanaRequest,
     response: LifecycleResponseFactory,
     toolkit: OnPreAuthToolkit
   ) {
+    const serverBasePath: string = getLegacyAPI().legacyConfig.get('server.basePath');
```

Comment:
Can't we just make `serverBasePath` property of BasePath service as `public readonly` to not confuse it with "request" base path, but at the same time still keep them together?

Having access to `kibana.index` is listed [as a dependency for the Security plugin as well](https://github.com/elastic/kibana/issues/33775). If it stays in Kibana plugin, then we need just Kibana NP plugin that would expose that for the time being (like we did with Timelion NP plugin and `timelion.ui.enabled` config key).

Reply:
> Can't we just make serverBasePath property of BasePath service as public readonly to not confuse it with "request" base path, but at the same time still keep them together?

Yep, that's a good idea!

> Having access to kibana.index is listed as a dependency for the Security plugin as well. If it stays in Kibana plugin, then we need just Kibana NP plugin that would expose that for the time being (like we did with Timelion NP plugin and timelion.ui.enabled config key).

Yeah that'd work too. So maybe the answer here is that plugins and services will just expose the properties as needed, rather than introducing a whitelisting system for the `yml` entries in general.

Reply:
I've created https://github.com/elastic/kibana/issues/45991